src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_sysfs.c

```diff
@@ -207,6 +207,13 @@ kds_echo_store(struct device *dev, struct device_attribute *da,
 }
 static DEVICE_ATTR(kds_echo, 0644, kds_echo_show, kds_echo_store);
 
+static ssize_t
+kds_mode_show(struct device *dev, struct device_attribute *attr, char *buf)
```

Comment:
Yeah, I know this is 'c' and not "c++", but how we handle the buffers really does concern me.

Reply:
Thanks to pointing this out. In Linux, this buffer has a 4KB limitation. I would fix potential buffer overrun issue for this two nodes that you mentioned. We should have another PR that clear up this potentially issue for other sysfs nodes like memstat, memstat_raw and etc.